StripeCryptoOnramp/StripeCryptoOnramp/Source/Components/CryptoOnrampCoordinator.swift

```diff
@@ -210,6 +223,50 @@ public final class CryptoOnrampCoordinator: CryptoOnrampCoordinatorProtocol {
             return nil
         }
     }
+
+    @MainActor
+    public func presentApplePay(using paymentRequest: PKPaymentRequest, from viewController: UIViewController) async throws -> ApplePayPaymentStatus {
+        try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<ApplePayPaymentStatus, Swift.Error>) in
+            guard let context = STPApplePayContext(paymentRequest: paymentRequest, delegate: self) else {
+                continuation.resume(throwing: ApplePayPaymentStatus.Error.applePayFallbackError)
+                return
+            }
+
+            // Retain the continuation until we receive a completion delegate callback.
+            self.applePayCompletionContinuation = continuation
+            context.presentApplePay()
+        }
+    }
+}
+
+extension CryptoOnrampCoordinator: STPApplePayContextDelegate {
+
+    // MARK: - STPApplePayContextDelegate
+
+    public func applePayContext(
+        _ context: STPApplePayContext,
+        didCreatePaymentMethod paymentMethod: STPPaymentMethod,
+        paymentInformation: PKPayment,
+        completion: @escaping STPIntentClientSecretCompletionBlock
+    ) {
+        self.applePayPaymentMethod = paymentMethod
+        completion(STPApplePayContext.COMPLETE_WITHOUT_CONFIRMING_INTENT, nil)
+    }
```

Comment:
Note that we store the payment method on this coordinator as outlined in this doc: https://docs.google.com/document/d/11aEvra9g3C3ixhc1opYpUKgUPRk06VFqjL5Sx88iL6g/edit?tab=t.0

We are not making further use of it yet.